src/main/java/works/crossdock/CrossdockClient.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package works.crossdock;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GenericFutureListener;
+import java.util.HashMap;
+import java.util.Map;
+import works.crossdock.client.Behavior;
+
+/**
+ * Runs crossdock behaviors. Behaviors are passed as input to the client. Callers explicitly contorl
+ * the start / stop of the client.
+ */
+public class CrossdockClient {
+  private final int crossdockPort;
+  private final Map<String, Behavior> behaviors;
+  private ChannelFuture future;
+
+  /**
+   * Creates a client handle for crossdock tests.
+   *
+   * @param crossdockPort port to run the crossdock server on
+   * @param inputBehaviors map of behaviors and actions to execute
+   */
+  public CrossdockClient(int crossdockPort, Map<String, Behavior> inputBehaviors) {
+    this.crossdockPort = crossdockPort;
+    this.behaviors = new HashMap<>(inputBehaviors);
+    this.future = null;
+  }
+
+  /**
+   * Starts the client by starting underlying http server.
+   *
+   * @throws Exception if server fails to start
+   */
+  public void start() throws Exception {
+    EventLoopGroup bossGroup = new NioEventLoopGroup();
+    EventLoopGroup workerGroup = new NioEventLoopGroup();
+    ServerBootstrap bootstrap = new ServerBootstrap();
+    bootstrap
+        .group(bossGroup, workerGroup)
+        .channel(NioServerSocketChannel.class)
+        .childHandler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              public void initChannel(SocketChannel ch) throws Exception {
+                ch.pipeline()
+                    .addLast(new HttpServerCodec())
+                    .addLast(new HttpObjectAggregator(Integer.MAX_VALUE))
+                    .addLast(new CrossdockServerInboundHandler(behaviors));
+              }
+            })
+        .option(ChannelOption.SO_BACKLOG, 128)
+        .childOption(ChannelOption.SO_KEEPALIVE, true);
+
+    future = bootstrap.bind("0.0.0.0", crossdockPort).sync();
```

Comment:
future is too generic. its like calling everything data, object, args and param.